api/client/client.go

```diff
@@ -325,7 +324,7 @@ func dialerConnect(ctx context.Context, params connectParams) (*Client, error) {
 		params.dialer = params.cfg.Dialer
 	}
 	clt := newClient(params.cfg, params.dialer, params.tlsConfig)
-	if err := clt.dialGRPC(ctx, constants.APIDomain); err != nil {
+	if err := clt.dialGRPC(ctx, ""); err != nil {
```

Comment:
Overall I find this PR very helpful in understanding what's going on.

This line, however, might still leave me wondering what's going on - any way we can clarify what it means to dial an empty addr?

Reply:
Added coments